src/main/java/openfasttrack/importer/ImporterFactory.java

```diff
@@ -1,14 +1,98 @@
 package openfasttrack.importer;
 
-import java.io.Reader;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toSet;
 
-import openfasttrack.importer.markdown.MarkdownImporter;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Set;
+import java.util.logging.Logger;
 
 /**
  * Factory for importers
  */
-public class ImporterFactory
+public abstract class ImporterFactory
 {
+    private final static Logger LOG = Logger.getLogger(ImporterFactory.class.getName());
+    private final Set<String> supportedFileExtensions;
+
+    protected ImporterFactory(final String... supportedFileExtensions)
+    {
+        this(asList(supportedFileExtensions));
+    }
+
+    protected ImporterFactory(final Collection<String> supportedFileExtensions)
+    {
+        this.supportedFileExtensions = supportedFileExtensions.stream() //
+                .map(String::toLowerCase) //
+                .collect(toSet());
+    }
+
+    /**
+     * Returns <code>true</code> if this {@link ImporterFactory} supports
+     * importing the given file based on its file extension.
+     *
+     * @param file
+     *            the file to check.
+     * @return <code>true</code> if the given file is supported for importing.
+     */
+    public boolean supportsFile(final Path file)
+    {
+        final String fileName = file.getFileName().toString().toLowerCase();
+        for (final String extension : this.supportedFileExtensions)
+        {
+            if (fileName.endsWith(extension))
```

Comment:
This is possibly dangerous. Imagine a situation where the registered extension is "zip" this would not only match the filename `archive.zip` but also a file called `unzip`.

Now that I think of it, filename patterns might be more explicit (and flexible) then extensions:

    {".*\.txt", "README", "readme"}
    {".*\.markdown", ".*\md"}

and so one